util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/SearchWorkCall.java

```diff
@@ -122,4 +123,16 @@ public String toString() {
 		return "search work execution on indexes '" + indexNames + "'; work = " + work;
 	}
 
+	static Integer getLoadingTimeout(StubSearchWork work) {
+		Long timeout = work.getFailAfterTimeout();
+		TimeUnit timeUnit = work.getFailAfterTimeUnit();
+		if ( timeout == null || timeUnit == null ) {
+			return null;
+		}
+
+		long nanos = timeUnit.toNanos( timeout );
+		int millis = Math.toIntExact( nanos / 1000000 );
+		return ( nanos % 1000000 == 0 ) ? millis : millis + 1;
```

Comment:
I believe this code was copy/pasted from the Lucene backend? Maybe it would be a good idea to move it to some shared util class.

Reply:
In Lucene backend we're using

``` java
timeoutUnit.toMillis( timeoutValue )
```
Probably it would be better doing the same here.

Reply:
Or maybe not, since the method handles null inputs and always perform a round up of the values.
Moving to an util class...

``` java
@Test
public void test() {
	long l = TimeUnit.NANOSECONDS.toMillis( 1L );
	Assertions.assertThat( l ).isEqualTo( 0 );

	Long aLong = ConvertUtils.toMilliseconds( 1L, TimeUnit.NANOSECONDS );
	Assertions.assertThat( aLong ).isEqualTo( 1L );
}
```